### PR TITLE
Update nvidia device plugin image to latest from upstream

### PR DESCRIPTION
**Pull request by xuto2**
Also to address CVE issue found in the old images

### CODE REVIEW

File: vhdbuilder/packer/install-dependencies.sh [hunk 1]
```diff
@@ -293,16 +293,10 @@ done
...
+v0.9.0
 "
 for NVIDIA_DEVICE_PLUGIN_VERSION in ${NVIDIA_DEVICE_PLUGIN_VERSIONS}; do
-  if [[ "${cliTool}" == "ctr" ]]; then
-    # containerd/ctr doesn't auto-resolve to docker.io
-    CONTAINER_IMAGE="docker.io/nvidia/k8s-device-plugin:${NVIDIA_DEVICE_PLUGIN_VERSION}"
-  else
-    CONTAINER_IMAGE="nvidia/k8s-device-plugin:${NVIDIA_DEVICE_PLUGIN_VERSION}"
-  fi
+    CONTAINER_IMAGE="mcr.microsoft.com/oss/nvidia/k8s-device-plugin:${NVIDIA_DEVICE_PLUGIN_VERSION}"
```
Comment (xuto2) [inline vhdbuilder/packer/install-dependencies.sh:299]: tested locally it works for both docker and containerd

File: vhdbuilder/packer/install-dependencies.sh [hunk 2]
```diff
@@ -293,16 +293,10 @@ done
 
 if [[ $OS == $UBUNTU_OS_NAME ]]; then
 NVIDIA_DEVICE_PLUGIN_VERSIONS="
-1.11
-1.10
+v0.9.0
```
Comment (eriksywu) [inline vhdbuilder/packer/install-dependencies.sh:296]: super nit: it seems like the pattern for install-dependencies is to append the "v" explicitly when constructing the image/pkg url, not in the VERSIONS array

File: vhdbuilder/packer/install-dependencies.sh [hunk 3]
```diff
@@ -316,9 +310,9 @@ if grep -q "fullgpu" <<< "$FEATURE_FLAGS" && grep -q "gpudaemon" <<< "$FEATURE_F
   DEST="/usr/local/nvidia/bin"
   mkdir -p $DEST
   if [[ "${CONTAINER_RUNTIME}" == "containerd" ]]; then
-    ctr --namespace k8s.io run --rm --mount type=bind,src=${DEST},dst=${DEST},options=bind:rw --cwd ${DEST} "docker.io/nvidia/k8s-device-plugin:1.11" plugingextract /bin/sh -c "cp /usr/bin/nvidia-device-plugin $DEST" || exit 1   
+    ctr --namespace k8s.io run --rm --mount type=bind,src=${DEST},dst=${DEST},options=bind:rw --cwd ${DEST} "mcr.microsoft.com/oss/nvidia/k8s-device-plugin:v0.9.0" plugingextract /bin/sh -c "cp /usr/bin/nvidia-device-plugin $DEST" || exit 1
```
Comment (eriksywu) [inline vhdbuilder/packer/install-dependencies.sh:313]: just curious why did we download both 1.10 and 1.11 before if we only ever install 1.11

Comment (xuto2) [reply vhdbuilder/packer/install-dependencies.sh:313]: just a legacy code and we though caching extra image there will help customers. Now with the CVE on these images, we shouldn't keep them anyways.

**Pull request merged**